pkg/logger/worker.go

```diff
@@ -75,85 +84,145 @@ func NewWorker(id int, workerQueue chan chan LogRequest, logger *zap.SugaredLogg
-	t, err := cloudevents.NewHTTP(
-		cloudevents.WithTarget(logReq.Url.String()),
-	)
+// prepareEvent creates a cloudevent from the LogRequest setting the extentions
+// body and metadata. Event defaulter functions should be applied here since
+// a cloudevent client is not used.
+func (w *Worker) prepareEvent(logReq LogRequest) (cloudevents.Event, error) {
+	event := cloudevents.NewEvent(cloudevents.VersionV1)
+	event.SetID(logReq.Id)
+	event.SetType(logReq.ReqType)
 
+	event.SetExtension(InferenceServiceAttr, logReq.InferenceService)
+	event.SetExtension(NamespaceAttr, logReq.Namespace)
+	event.SetExtension(ComponentAttr, logReq.Component)
+	event.SetExtension(EndpointAttr, logReq.Endpoint)
+
+	encodedMetadata, err := json.Marshal(logReq.Metadata)
 	if err != nil {
-		return fmt.Errorf("while creating http transport: %w", err)
+		return event, fmt.Errorf("could not encode metadata as json: %w", err)
+	}
+	event.SetExtension(MetadataAttr, string(encodedMetadata))
+
+	event.SetSource(logReq.SourceUri.String())
+	if err := event.SetData(logReq.ContentType, *logReq.Bytes); err != nil {
+		return event, fmt.Errorf("while setting cloudevents data: %w", err)
+	}
+
+	if event.Time().IsZero() {
+		event.SetTime(time.Now())
 	}
+	return event, nil
+}
 
-	if logReq.Url.Scheme == "https" {
-		caCertFilePath := filepath.Join(LoggerCaCertMountPath, logReq.CertName)
+// buildClient constructs an http client and transport to send cloudevent request.
+func (w *Worker) buildClient(url *url.URL, certName string, tlsSkipVerify bool) (*http.Client, error) {
+	c := &http.Client{}
+
+	if url.Scheme == "https" {
+		caCertFilePath := filepath.Join(LoggerCaCertMountPath, certName)
 		caCertFile, err := os.ReadFile(caCertFilePath)
 		// Do not fail if certificates not found, for backwards compatibility
 		if err == nil {
 			clientCertPool := x509.NewCertPool()
 			if !clientCertPool.AppendCertsFromPEM(caCertFile) {
-				return fmt.Errorf("while parsing CA certificate")
+				return nil, fmt.Errorf("while parsing CA certificate")
 			}
 
 			tlsTransport := &http.Transport{
 				TLSClientConfig: &tls.Config{
 					RootCAs:            clientCertPool,
 					MinVersion:         tls.VersionTLS12,
-					InsecureSkipVerify: logReq.TlsSkipVerify, // #nosec G402
+					InsecureSkipVerify: tlsSkipVerify, // #nosec G402
 				},
 			}
-			t.Client.Transport = tlsTransport
+			c.Transport = tlsTransport
 		} else {
 			w.Log.Warnf("using https endpoint but could not find CA cert file %s", caCertFilePath)
 		}
 	}
+	return c, nil
+}
 
-	c, err := cloudevents.NewClient(t,
-		cloudevents.WithTimeNow(),
-	)
+func (w *Worker) sendBatchCloudEvent(logReqs []LogRequest) error {
+	events := make([]cloudevents.Event, 0, len(logReqs))
+
+	for _, lr := range logReqs {
+		event, err := w.prepareEvent(lr)
+		if err != nil {
+			return err
+		}
+		events = append(events, event)
+	}
+
+	c, err := w.buildClient(logReqs[0].Url, logReqs[0].CertName, logReqs[0].TlsSkipVerify)
```

Comment:
These fields looks like should be common to all the log requests

Reply:
I agree these fields are shared and infact they are static through the lifetime of the logger. The current implementation creates a client and loads the certs each time it sends a log request. Not only are we sending more data over the channel, but we are not re-using the client connections. 

I debated refactoring this in this PR, but it would likely require some larger changes to the dispatcher, worker and handler. I think including it in this PR would make the change too large and complicate the logic. I think its worth while doing that refactor in a separate PR.